lighthouse/multi-cluster-service-selection.md

```diff
@@ -0,0 +1,71 @@
+# Multi-Cluster Service Selection Optimization - EP
+
+## Motivation
+
+While the world is moving towards micro-services application based architecture, more organizations combine multi-cluster and multi-cloud providers for redundancy, performance, management, and other reasons.
+We would like to optimize the total cost of a Kubernetes deployment in means of communication between multi-clusters, given that we could decide between different approaches. The three approaches possible, currently, are optimizing the full service chain of a single app/deployment, multiple apps/deployments, or a smart local optimization per cluster point of view, without building the full service chain.
+In this Enhancement Proposal, we shall introduce questions, ideas, and requirements to optimize the service selection mechanism of the Lighthouse Plugin.
+
+
+### Requirements
+1. Topology:
+    1. Selection Decision should be made locally, i.e per cluster - decentralized solution.
+    2. The solution must be adaptive for topology changes.
+1. Cost Parameters/Metrics:
+    1. Latency
+    2. RTT
+    3. Networking cost
+    4. Network congestion
+    5. CPU/RAM load on each container/pod
+    6. Service scale/autoscale (which actually will affect the load on the CPU and RAM)
+
+
+### Open Questions
+1. Topology:
+    1. What topology would we like to optimize?
+        1. Local optimization per cluster?
+        2. Per App optimization?
+        3. Multi App optimization?
+    2. How often do deployments affects changes in the topology?
+    3. How adaptive would we like to be for changes in the topology?
+    4. Would we like to build the full application graph for better optimization?

```

Comment:
Great point, this is exactly the kind of insights we need to discuss... would you say that knowing the application graph or parts of it (enable look ahead for example) under any conditions will be something we would like to avoid?